tensorflow_encrypted/protocol/securenn.py

```diff
@@ -40,8 +40,11 @@ def __init__(
             **kwargs
         )
         self.server_2 = server_2
-        self.prime_factory = prime_factory or PrimeFactory(107)
+
+        self.prime_factory = prime_factory or PrimeFactory(107, native_type=self.tensor_factory.native_type)
```

Comment:
I've been thinking about this part of using `kwarg=None` in combination with `x = kwarg or default_value`. I think it's less readable and I think we should change these kinds of antipatterns to: 
```
if kwarg is not None:
     x = kwarg
else:
     ...
```

Reply:
I agree it's not super easy to read